File: src/server/src/main/java/com/project/domain/image/PostImage.java
package com.project.domain.image;

import com.project.domain.post.Post;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;

@Entity(name = "post_images")
public class PostImage extends Image {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id", nullable = false)
    private Post post;

    protected PostImage() {
    }

    public PostImage(String url, Post post) {
        super(url);
        this.post = post;
    }

    public Post getPost() {
        return post;
    }
}

File: src/server/src/main/kotlin/com/project/domain/image/PostImage.kt
package com.project.domain.image

import com.project.domain.post.Post
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne

@Entity(name = "post_images")
class PostImage : Image {
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id", nullable = false)
    var post: Post? = null
        private set

    protected constructor()

    constructor(url: String?, post: Post?) : super(url) {
        this.post = post
    }
}
